Index: src/contracts/extensions/stata-token/interfaces/IERC4626StataToken.sol
===================================================================
--- src/contracts/extensions/stata-token/interfaces/IERC4626StataToken.sol
+++ src/contracts/extensions/stata-token/interfaces/IERC4626StataToken.sol
@@ -35,10 +35,9 @@
      * @notice Burns `shares` of static aToken, with receiver receiving the corresponding amount of aToken
      * @param shares The shares to withdraw, in static balance of StaticAToken
      * @param receiver The address that will receive the amount of `ASSET` withdrawn from the Aave protocol
      * @return amountToWithdraw: aToken send to `receiver`, dynamic balance
-     *
-     */
+     **/
     function redeemATokens(
         uint256 shares,
         address receiver,
         address owner
@@ -48,10 +47,9 @@
      * @notice Deposits aTokens and mints static aTokens to msg.sender
      * @param assets The amount of aTokens to deposit (e.g. deposit of 100 aUSDC)
      * @param receiver The address that will receive the static aTokens
      * @return uint256 The amount of StaticAToken minted, static balance
-     *
-     */
+     **/
     function depositATokens(
         uint256 assets,
         address receiver
     ) external returns (uint256);
@@ -62,10 +60,9 @@
      * @param receiver The address that will receive the static aTokens
      * @param deadline Must be a timestamp in the future
      * @param sig A `secp256k1` signature params from `msgSender()`
      * @return uint256 The amount of StaticAToken minted, static balance
-     *
-     */
+     **/
     function depositWithPermit(
         uint256 assets,
         address receiver,
         uint256 deadline,
