Index: src/contracts/interfaces/IInitializableAToken.sol
===================================================================
--- src/contracts/interfaces/IInitializableAToken.sol
+++ src/contracts/interfaces/IInitializableAToken.sol
@@ -34,21 +34,17 @@
 
     /**
      * @notice Initializes the aToken
      * @param pool The pool contract that is initializing this contract
-     * @param treasury The address of the Aave treasury, receiving the fees on this aToken
      * @param underlyingAsset The address of the underlying asset of this aToken (E.g. WETH for aWETH)
-     * @param incentivesController The smart contract managing potential incentives distribution
      * @param aTokenDecimals The decimals of the aToken, same as the underlying asset's
      * @param aTokenName The name of the aToken
      * @param aTokenSymbol The symbol of the aToken
      * @param params A set of encoded parameters for additional initialization
      */
     function initialize(
         IPool pool,
-        address treasury,
         address underlyingAsset,
-        IAaveIncentivesController incentivesController,
         uint8 aTokenDecimals,
         string calldata aTokenName,
         string calldata aTokenSymbol,
         bytes calldata params
