===================================================================
--- src/core/contracts/protocol/libraries/aave-upgradeability/BaseImmutableAdminUpgradeabilityProxy.sol
+++ src/contracts/misc/aave-upgradeability/BaseImmutableAdminUpgradeabilityProxy.sol
@@ -1,8 +1,8 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.10;
 
-import {BaseUpgradeabilityProxy} from "../../../dependencies/openzeppelin/upgradeability/BaseUpgradeabilityProxy.sol";
+import {BaseUpgradeabilityProxy} from "../../dependencies/openzeppelin/upgradeability/BaseUpgradeabilityProxy.sol";
 
 /**
  * @title BaseImmutableAdminUpgradeabilityProxy
  * @author Aave, inspired by the OpenZeppelin upgradeability proxy pattern
@@ -17,12 +17,12 @@
     address internal immutable _admin;
 
     /**
      * @dev Constructor.
-     * @param admin The address of the admin
+     * @param admin_ The address of the admin
      */
-    constructor(address admin) {
-        _admin = admin;
+    constructor(address admin_) {
+        _admin = admin_;
     }
 
     modifier ifAdmin() {
         if (msg.sender == _admin) {
