===================================================================
--- src/core/contracts/protocol/configuration/PoolAddressesProvider.sol
+++ src/contracts/protocol/configuration/PoolAddressesProvider.sol
@@ -1,10 +1,10 @@
-// SPDX-License-Identifier: MIT
+// SPDX-License-Identifier: BUSL-1.1
 pragma solidity ^0.8.10;
 
 import {Ownable} from "../../dependencies/openzeppelin/contracts/Ownable.sol";
 import {IPoolAddressesProvider} from "../../interfaces/IPoolAddressesProvider.sol";
-import {InitializableImmutableAdminUpgradeabilityProxy} from "../libraries/aave-upgradeability/InitializableImmutableAdminUpgradeabilityProxy.sol";
+import {InitializableImmutableAdminUpgradeabilityProxy} from "../../misc/aave-upgradeability/InitializableImmutableAdminUpgradeabilityProxy.sol";
 
 /**
  * @title PoolAddressesProvider
  * @author Aave
